simplified for loop

MCWertGaming opened the pull request:
As introduced in #64 and implemented in #68: This pull-request tries to simplify the get_cursor_position() function. The code should be checked of logic mistakes. I have tried to just change the code, but don't touch the logic at all. I hope it's still intakt!

certik commented:
The fact that even you forgot to append `std::flush` and had to fix it in 4bb0f822d10f91c7dd9392d198e629df08d9b87d is an argument against #81.

MCWertGaming commented:
the fix in 4bb0f82 was due to a misstake in the old pull-request #68, I made the misstake there. I'll have to resolve that conflict in git, but will do that, after #81 was merged of closed. Like I said both is absolutely fine! I just did brain dead copy paste here, sorry for that!
I'll change it to the write() function, if you decide to drop #81.

MCWertGaming commented:
Because we dropped #81 and changed it to #87, I'll wait for #87 to get merged and then change the std::cout here back to the write function. You can take a look at the logic then @certik.

certik commented:
#87 is merged, you can go ahead and finish the PR now.

MCWertGaming commented:
done!

certik commented on cpp-terminal/terminal.h:
```diff
@@ -381,34 +381,31 @@ class Terminal: public BaseTerminal {
     }
 
-    void get_cursor_position(int& rows, int& cols) const
+void get_cursor_position(int& rows, int& cols) const
```
```suggestion
    void get_cursor_position(int& rows, int& cols) const
```

certik commented on cpp-terminal/terminal.h:
```diff
@@ -381,34 +381,31 @@ class Terminal: public BaseTerminal {
-        while (i < sizeof(buf) - 1) {
-            while (!read_raw(&buf[i])) {
-            };
+        for (unsigned int i = 0; i < sizeof(buf) -1; i++) {
```
```suggestion
        for (unsigned int i = 0; i < sizeof(buf) - 1; i++) {
```

certik commented on cpp-terminal/terminal.h:
```diff
@@ -381,34 +381,31 @@ class Terminal: public BaseTerminal {
-            };
+        for (unsigned int i = 0; i < sizeof(buf) -1; i++) {
+            while (!read_raw(&buf[i]));
             if (buf[i] == 'R')
+            {
```
```suggestion
            if (buf[i] == 'R') {
```

certik commented on cpp-terminal/terminal.h:
```diff
@@ -381,34 +381,31 @@ class Terminal: public BaseTerminal {
+            while (!read_raw(&buf[i]));
             if (buf[i] == 'R')
+            {
+                if (i < 5)
+                    throw std::runtime_error("get_cursor_position(): too short response");
+                else
+                    buf[i] = '\0';
```
```suggestion
                if (i < 5) {
                    throw std::runtime_error("get_cursor_position(): too short response");
                } else {
                    buf[i] = '\0';
                }
```

certik commented on cpp-terminal/terminal.h:
```diff
@@ -381,34 +381,31 @@ class Terminal: public BaseTerminal {
             }
-            if (buf[i] == '\0') break;
-            i++;
+            if (buf[i] == '\0') 
+                break;
```
When you have multineline if / while statements, use `{}` to prevent a mistake in the future:
```suggestion
            if (buf[i] == '\0') {
                break;
            }
```
Or put them on the same line:
```suggestion
            if (buf[i] == '\0') break;
```

MCWertGaming replied:
about the {}, that's true, i sometimes forget that. There is a compiler waning on that btw.

certik reviewed (changes requested):
I think it looks good. It seems it is equivalent. I left some minor formatting suggestions.

MCWertGaming commented:
about the formating: the code base has no real coding style. Sometimes I have seen 2 spaces, sometimes 4 spaces, sometimes a new line after curly brackets, sometimes there is none. We should really use clang format for that. Then there wont be such things to check anymore.

certik reviewed (approved):
I think this looks good.
